modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -2636,6 +2904,141 @@ else if (selectExpression instanceof ScalarDSLQueryAlias) {
+			_createInsertIntoLocalizationTableStatement(
+				dynamicObjectDefinitionLocalizationTable, values);
+
+		if (insertIntoLocalizationTableStatement == null) {
+			return;
+		}
+
+		Set<Locale> locales = new HashSet<>();
+
+		for (ObjectField objectField :
+				dynamicObjectDefinitionLocalizationTable.getObjectFields()) {
+
+			Map<String, String> languageIdValues =
+				(Map<String, String>)values.get(
+					objectField.getI18nObjectFieldName());
+
+			if (languageIdValues == null) {
+				continue;
+			}
+
+			for (String languageId : languageIdValues.keySet()) {
+				locales.add(LocaleUtil.fromLanguageId(languageId));
+			}
+		}
+
+		locales = SetUtil.intersect(
+			locales,
+			_language.getCompanyAvailableLocales(
+				objectDefinition.getCompanyId()));
+
+		Connection connection = _currentConnection.getConnection(
+			objectEntryPersistence.getDataSource());
+
+		for (Locale locale : locales) {
+			_executeInsertIntoLocalizationTable(
+				connection, dynamicObjectDefinitionLocalizationTable,
+				insertIntoLocalizationTableStatement, values, locale,
+				objectEntryId,
+				dynamicObjectDefinitionLocalizationTable.getObjectFields());
+		}
+	}
+
+	private void _insertIntoOrUpdateLocalizationTable(
+			ObjectDefinition objectDefinition, long objectEntryId,
+			Map<String, Serializable> values)
+		throws PortalException {
+
+		DynamicObjectDefinitionLocalizationTable
+			dynamicObjectDefinitionLocalizationTable =
+				_dynamicObjectDefinitionLocalizationTableFactory.create(
+					objectDefinition);
+
+		if (dynamicObjectDefinitionLocalizationTable == null) {
+			return;
+		}
+
+		List<String> objectEntryPersistedLanguageIds = new ArrayList<>(
+			objectEntryPersistence.dslQuery(
+				DSLQueryFactoryUtil.select(
+					dynamicObjectDefinitionLocalizationTable.
+						getLanguageIdColumn()
+				).from(
+					dynamicObjectDefinitionLocalizationTable
+				).where(
+					dynamicObjectDefinitionLocalizationTable.
+						getForeignKeyColumn(
+						).eq(
+							objectEntryId
+						)
+				)));
+
+		Map<String, Serializable> newI18nObjectFieldValues = new HashMap<>();
+
+		for (ObjectField objectField :
+				dynamicObjectDefinitionLocalizationTable.getObjectFields()) {
+
+			if (!values.containsKey(objectField.getI18nObjectFieldName())) {
+				continue;
+			}
+
+			Map<String, String> languageIdValues =
+				(Map<String, String>)values.get(
+					objectField.getI18nObjectFieldName());
+
+			for (Map.Entry<String, String> entry :
+					languageIdValues.entrySet()) {
+
+				if (objectEntryPersistedLanguageIds.contains(entry.getKey())) {
+					_updateLocalizationTable(
+						dynamicObjectDefinitionLocalizationTable,
+						entry.getKey(), objectEntryId,
+						HashMapBuilder.<String, Serializable>put(
+							objectField.getI18nObjectFieldName(),
+							HashMapBuilder.put(
+								entry.getKey(), entry.getValue()
+							).build()
+						).build());
+
+					continue;
+				}
+
+				newI18nObjectFieldValues.computeIfAbsent(
```

Comment:
Try to use compute here

Reply:
Done [here](https://github.com/liferay-objects/liferay-portal/pull/1913/commits/70fa88259d255cc53baa89eee9fa67be156528be)